### ISSUE TITLE
Attach image to a card, stored in nextcloud not possible (crash)

### ISSUE THREAD

**Issue by velletti** [stefan-niedermann/nextcloud-deck#814]
1. Go to ' single card -> attachment'
2. Click on ' show all files '
3. Scroll to your nextcloud server. select the folder with image and select image'

```
App Version: 1.13.4
App Version Code: 1013004
Server App Version: 1.2.3
App Flavor: play

Files App Version Code: 30140390

---

OS Version: 4.14.116(10.1.0.150C431)
OS API Level: 29
Device: HWVOG
Manufacturer: HUAWEI
Model (and Product): VOG-L29 (VOG-L29EEA)

---

java.io.FileNotFoundException: Error downloading file: Speedtest1.png
	at android.database.DatabaseUtils.readExceptionWithFileNotFoundExceptionFromParcel(DatabaseUtils.java:149)
	at android.content.ContentProviderProxy.openTypedAssetFile(ContentProviderNative.java:705)
	at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:1698)
	at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:1514)
	at android.content.ContentResolver.openInputStream(ContentResolver.java:1198)
	at it.niedermann.nextcloud.deck.util.AttachmentUtil.copyContentUriToTempFile(AttachmentUtil.java:79)
	at it.niedermann.nextcloud.deck.ui.card.attachments.CardAttachmentsFragment.uploadNewAttachmentFromUri(CardAttachmentsFragment.java:415)
	at it.niedermann.nextcloud.deck.ui.card.attachments.CardAttachmentsFragment.onActivityResult(CardAttachmentsFragment.java:382)
	at androidx.fragment.app.FragmentActivity.onActivityResult(FragmentActivity.java:170)
	at android.app.Activity.dispatchActivityResult(Activity.java:8412)
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5580)
	at android.app.ActivityThread.handleSendResult(ActivityThread.java:5628)
	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:51)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:149)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:103)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2473)
	at android.os.Handler.dispatchMessage(Handler.java:110)
	at android.os.Looper.loop(Looper.java:219)
	at android.app.ActivityThread.main(ActivityThread.java:8347)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1055

```

**Expected behavior**
Image should be attaced like images i took with my phone and are stored locally on the phone. 
if i first download the same image to the phone f.e. by sending it via whatsapp or put it to gallery,, same image works.

or show message -> first download to to phone .... 

**Versions**
 - Nextcloud: 20.0.5 
 - Nextcloud Deck: 
 - Nextcloud Android: 
 - Nextcloud Android Deck: 

**Smartphone (please complete the following information):**
- Device: Huawei P30 Pro
- Android-Version:  10
- App-Store:
  - [x] Google Play Store

**Are you using LDAP?**
- [x] No

**Comment by stefan-niedermann** [comment]
Absolutel, thank you for the report 🙂
I'll have a look at this.

**Comment by stefan-niedermann** [comment]
That's a bit weird: I am able to reproduce this on my physical device but on the emulator it works as expected. I will try to debug my physical device now to find the differences.

**Comment by stefan-niedermann** [comment]
Okay, it looks like picking this kind of files must not happen on the UI thread (which makes sense because it might not yet be available on the device and a download would block the UI).

The fix will be released with version `1.14.2` which will soon be available at F-Droid and the [Play Store beta channel](https://play.google.com/apps/testing/it.niedermann.nextcloud.deck.play). It will unfortunately take some more time to publish this version at the Play Store main channel for several reasons. Please be a bit more patient or switch to the beta channel.

Original hint which guided me to the solution: https://help.nextcloud.com/t/android-app-select-file-with-nextcloud-app-file-cant-be-read/103706

**Comment by stefan-niedermann** [comment]
Thanks for the report :heart: 

Feel free to reopen the issue in case it occurs again *after* `1.14.2`.
